api/views.py

```diff
@@ -1,18 +1,19 @@
 from django.http import JsonResponse
 from django.contrib.auth.decorators import login_required
-
-from oauth2_provider.views import ScopedProtectedResourceView
+from oauth2_provider.views import TokenView
 
 from core.decorators import class_decorator
 
 
 @class_decorator(login_required)
-class ProfileView(ScopedProtectedResourceView):
-    required_scopes = ['read']
+class ProfileView(TokenView):
 
     def get(self, request, *args, **kwargs):
         user = request.user
-        return JsonResponse({
-            'id': user.pk,
-            'email': user.email,
-        })
+        _, request = self.verify_request(request)
+        response = {'id': user.pk}
+
+        if request.access_token.scope == 'email':
```

Comment:
I'm not sure where we want to go here in terms of abstraction/customization. I guess that this is not the default behavior that we should push upstream, still it's interesting to be able to explicitly define somewhere (settings?) the relations scopes ⟷ returned fields. We can do that in another PR but it's something we should not forget. Thoughts?